### PR TITLE
Rewrite index.js to be more readable

### PR DESCRIPTION
**Pull request by NicholasEllul**
I came across https://github.com/kkomelin/isomorphic-dompurify/issues/10 where others were trying to understand index.js and so I made cosmetic changes (renaming variables and moving things into methods) without changing the flow of logic so people have a more clear idea.

### Some thoughts I need your opinion on:

* While I did refactor this code to be logically identical to what was there before it, there of course is risk involved. If you believe the risk of this change outweighs the benefits I am happy to close this PR.
* Since this was not a DOMPurify change, I made this change a patch version change. However, to keep consistency I can change this to a minor if you prefer.

### CODE REVIEW

**Comment by kkomelin** [comment]
Hi @NicholasEllul Thanks for the PR. Will review it.

The first thing I'd like to point attention to is why the variables and functions were named that way. I think the goal was to keep the size small, especially for the browser.

Of course, you can always obfuscate/pack the code on build but it would require additional build step.

**Comment by NicholasEllul** [comment]
> I think the goal was to keep the size small, especially for the browser to be almost invisible adapter between the browser

Fair point that is another reason to not perform the change. If there is no value in improving readability we can close this 👍

**Comment by kkomelin** [comment]
Hey @NicholasEllul,
Overall, it's a good idea to make the code more readable/maintainable, so can we please keep this issue open for some time? 

I'd like to experiment with the idea of obfuscating/packing your code with a console command every time when it's changed, for example [tcs](https://www.typescriptlang.org/docs/handbook/compiler-options.html) (not sure yet if it's possible with tcs).
So we would have `index.ts` file which we would pack into `index.js` with shorter function names and without blank spaces, comments, etc. Do you think it makes sense?

**Comment by NicholasEllul** [comment]
Great idea, yes if we can do that and only publish the obfuscated version to npm then we could have the best of both worlds 👍 I'll investigate why this is breaking CI as well as from my local tests I did not experience this

File: index.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   const { window } = new JSDOM('<!DOCTYPE html>');
-  return DOMPurifyInitializer(window)
+  return DOMPurifyInitializer(window);
 }
 
-function resolveDOMPurify(){
+function resolveDOMPurify() {
   const isClientSide = typeof process === 'undefined';
-  isClientSide ? importModule(require('dompurify')) : initDOMPurifyWithJSDOM();
+  return isClientSide ? importModule(require('dompurify')) : initDOMPurifyWithJSDOM();
```
Comment (kkomelin) [inline index.js:14]: @NicholasEllul I've just added the return statement to fix tests.

**Comment by kkomelin** [comment]
Hi @NicholasEllul,

I've added a build command that packs the source code. Surprisingly, the resulting code looks very similar to the original code (before your readability changes). Unfortunately, the `tcs` command doesn't pack the code properly, so I used [terser](https://www.npmjs.com/package/terser) instead. So no TypeScript for now.

Also, I suggest increasing the patch version for this change because it's not a bug fix, and it's rater a feature.

Please take a look when you have time.

File: index.js (hunk 1 above)
Comment (NicholasEllul) [reply index.js]: Haha this is how you can tell i've been writing too much ruby lately, thanks

**Comment by NicholasEllul** [comment]
Tested it and things look great on my end. Nice job with terser

File: index.js (hunk 1 above)
Comment (kkomelin) [reply index.js]: Ah, yes, Ruby automatically returns the result of the last statement.

**Comment by kkomelin** [comment]
@NicholasEllul 

> Also, I suggest increasing the patch version for this change because it's not a bug fix, and it's rater a feature.

Sorry, I mistakenly suggested a patch version but I actually meant a minor one. Hope you don't mind of releasing it as minor.

**Review by kkomelin — APPROVED** [review]
Thank you @NicholasEllul for this addition! 
I think we're almost ready for a new release. I'll soon release the new minor version - 0.12.0.

**Comment by kkomelin** [comment]
Released https://github.com/kkomelin/isomorphic-dompurify/releases/tag/v0.12.0

**Pull request merged**